src/mrinufft/operators/base.py

```diff
@@ -531,6 +531,14 @@ def norm_factor(self):
         """Normalization factor of the operator."""
         return np.sqrt(np.prod(self.shape) * (2 ** len(self.shape)))
 
+    def check_shape(self, image, init_shape):
+        """Check if the image shape is compatible with the operator's init shape."""
+        if image.shape != init_shape:
+            raise ValueError(
+                f"Image shape {image.shape} is not compatible with the operator shape"
+                f" {self.shape}"
+            )
+
```

Comment:
```suggestion
    def check_shape(self, image):
        """Check if the image shape is compatible with the operator's init shape."""
        if image.shape[-len(self.shape):] != self.shape:
            raise ValueError(
                f"Image shape {image.shape} is not compatible with the operator shape"
                f" {self.shape}"
            )

```

Reply:
Yes, you're right, that's why the test failed.